crates/oxc_minifier/src/compressor/mod.rs

```diff
@@ -228,6 +240,36 @@ impl<'a> Compressor<'a> {
         }
     }
 
+    /// Transforms variable declarations with `undefined` assignments into
+    /// declaration-only statements.
+    ///
+    /// > _note: cannot use this on `const` declarations. `const x;` is not valid._
+    ///
+    /// `let x = undefined; -> let x`
+    ///
+    /// Returns `true` if compression (mutation) occurred.
+    fn compress_declare_undefined<'b>(&mut self, stmt: &'b mut VariableDeclarator<'a>) -> bool {
+        // short circuit if we don't need compression. Helps avoid borrow
+        // checker violations when mutating
+        match &stmt.init {
+            // todo: handle `void 0` also?
```

Comment:
If you want to handle `void 0`, you may need to consider [`try_reduce_void`](https://github.com/Boshen/oxc/blob/9e24dbb0fa3feef155c564213201850e93b83a5f/crates/oxc_minifier/src/compressor/fold.rs#L511). Should you change the order of this function call in `visit_variable_declaration` ?🤔